tests/integration/cmd_link_unlink_test.go

```diff
@@ -9,138 +13,199 @@ import (
 
-	/*
-		Uncomment when we uncomment the test specs
-		var frontendContext, backendContext string
-		var oc helper.OcRunner
-	*/
-
 	var commonVar helper.CommonVar
 
-	// This is run before every Spec (It)
 	var _ = BeforeEach(func() {
-		// oc = helper.NewOcRunner("oc")
 		commonVar = helper.CommonBeforeEach()
+		// wait until timeout(sec) for odo to see all the operators installed by setup script in the namespace
+		odoArgs := []string{"catalog", "list", "services"}
+		operator := "service-binding-operator"
+		helper.WaitForCmdOut("odo", odoArgs, 5, true, func(output string) bool {
+			return strings.Contains(output, operator)
+		})
 	})
 
-	// Clean up after the test
-	// This is run after every Spec (It)
 	var _ = AfterEach(func() {
 		helper.CommonAfterEach(commonVar)
 	})
 
-	Context("when running help for link and unlink command", func() {
-		It("should display the help", func() {
+	It("should display the help", func() {
+		By("for the link command", func() {
 			appHelp := helper.Cmd("odo", "link", "-h").ShouldPass().Out()
-			helper.MatchAllInOutput(appHelp, []string{"Link component to a service ", "backed by an Operator or Service Catalog", "or component", "works only with s2i components"})
-			appHelp = helper.Cmd("odo", "unlink", "-h").ShouldPass().Out()
+			helper.MatchAllInOutput(appHelp, []string{"Link component to a service ", "backed by an Operator or Service Catalog", "or component"})
+		})
+		By("for the unlink command", func() {
+			appHelp := helper.Cmd("odo", "unlink", "-h").ShouldPass().Out()
 			Expect(appHelp).To(ContainSubstring("Unlink component or service from a component"))
 		})
 	})
 
-	/*
-		Context("When link between components using wrong port", func() {
-			JustBeforeEach(func() {
-				frontendContext = helper.CreateNewContext()
-				backendContext = helper.CreateNewContext()
-			})
-			JustAfterEach(func() {
-				helper.DeleteDir(frontendContext)
-				helper.DeleteDir(backendContext)
-			})
-			It("should fail", func() {
-				helper.CopyExample(filepath.Join("source", "nodejs"), frontendContext)
-				helper.Cmd("odo", "create", "--s2i", "nodejs", "frontend", "--context", frontendContext, "--project", commonVar.Project).ShouldPass()
-				helper.Cmd("odo", "push", "--context", frontendContext).ShouldPass()
-				helper.CopyExample(filepath.Join("source", "python"), backendContext)
-				helper.Cmd("odo", "create", "--s2i", "python", "backend", "--context", backendContext, "--project", commonVar.Project).ShouldPass()
-				helper.Cmd("odo", "push", "--context", backendContext).ShouldPass()
-				stdErr := helper.Cmd("odo", "link", "backend", "--context", frontendContext, "--port", "1234").ShouldFail().Err()
-				Expect(stdErr).To(ContainSubstring("Unable to properly link to component backend using port 1234"))
-			})
+	When("two components are deployed", func() {
+		var frontendContext, backendContext, frontendURL, frontendComp, backendComp string
+		var runner helper.CliRunner
```

Comment:
IMO we can use `commonVar.CliRunner` directly, rather than introducing a new var. Correct me if I am wrong 🙂

Reply:
Oh shoot, I misunderstood it for `helper.CliRunner`. Will get to it.